### ISSUE TITLE
Tests broken on Darwin

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#16283]
Migrated from [rt.perl.org#132526](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=132526) (status was 'resolved')

Searchable as RT132526$

**Comment by p5pRT** [comment]
### From @dur-randir

#### Created by @dur-randir



Recent commit 184f90dc410cf3e66a5e682337ae54b3c9f9311f
Author&#8203;: Tony Cook \<tony@&#8203;develop\-help\.com>
Date&#8203;:   Fri Dec 1 01&#8203;:05&#8203;:07 2017 \+1100

&nbsp;    \(perl \#132506\) deal with un\-/partly implemented \*at\(\) functions

broke the following test&#8203;:

% \./perl \./harness run/switches\.t
run/switches\.t \.\. 47/137 \# Failed test 130 \- fail final rename at
\./test\.pl line 1074
\#      got 'Cannot complete in\-place edit of tmpinplace/foo&#8203;: failed to
rename work file \\'tmpinplace/SrDX3NRv\\' to \\'tmpinplace/foo\\'&#8203;:
Permission denied at \- line 5\, \<> line 1\.'
\# expected /\(?^&#8203;:Can't rename\)/
\# PROG&#8203;:
\# @&#8203;ARGV = \("tmpinplace/foo"\);
\# $^I = "";
\# while \(\<>\) \{
\#   chmod 0500\, "tmpinplace";
\#   print;
\# \}
\# print "ok\\n";
\# STATUS&#8203;: 3328
run/switches\.t \.\. Failed 1/137 subtests
&nbsp;        \(less 6 skipped subtests&#8203;: 130 okay\)

Test Summary Report
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
run/switches\.t \(Wstat&#8203;: 0 Tests&#8203;: 137 Failed&#8203;: 1\)
&nbsp;  Failed test&#8203;:  130
Files=1\, Tests=137\,  1 wallclock secs \( 0\.05 usr  0\.01 sys \+  0\.53
cusr  0\.60 csys =  1\.19 CPU\)
Result&#8203;: FAIL

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl 5.27.7:

Configured by dur-randir at Fri Dec  1 17:01:46 MSK 2017.

Summary of my perl5 (revision 5 version 27 subversion 7) configuration:
  Commit id: 3c67ad9b9e6d659feb76f3acfc9f81cfe59e3660
  Platform:
    osname=darwin
    osvers=13.4.0
    archname=darwin-2level
    uname='darwin isengard.local 13.4.0 darwin kernel version 13.4.0:
mon jan 11 18:17:34 pst 2016; root:xnu-2422.115.15~1release_x86_64
x86_64 '
    config_args='-de -Dusedevel -DDEBUGGING'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-fno-common -DPERL_DARWIN -mmacosx-version-min=10.9
-DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector
-I/usr/local/include -DPERL_USE_SAFE_PUTENV'
    optimize='-O3 -g'
    cppflags='-fno-common -DPERL_DARWIN -mmacosx-version-min=10.9
-DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector
-I/usr/local/include'
    ccversion=''
    gccversion='4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.56)'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -mmacosx-version-min=10.9 -fstack-protector -L/usr/local/lib'
    libpth=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/6.0/lib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/lib
/usr/local/lib /usr/lib
    libs=-lpthread -lgdbm -ldbm -ldl -lm -lutil -lc
    perllibs=-lpthread -ldl -lm -lutil -lc
    libc=
    so=dylib
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=bundle
    d_dlsymun=undef
    ccdlflags=' '
    cccdlflags=' '
    lddlflags=' -mmacosx-version-min=10.9 -bundle -undefined
dynamic_lookup -L/usr/local/lib -fstack-protector'



@INC for perl 5.27.7:
    lib
    /usr/local/lib/perl5/site_perl/5.27.7/darwin-2level
    /usr/local/lib/perl5/site_perl/5.27.7
    /usr/local/lib/perl5/5.27.7/darwin-2level
    /usr/local/lib/perl5/5.27.7


Environment for perl 5.27.7:
    DYLD_LIBRARY_PATH (unset)
    HOME=/Users/dur-randir
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/Users/dur-randir/perlbrew/bin:/Users/dur-randir/perlbrew/perls/perl-5.22.1/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/texbin
    PERLBREW_BASHRC_VERSION=0.78
    PERLBREW_HOME=/Users/dur-randir/.perlbrew
    PERLBREW_MANPATH=/Users/dur-randir/perlbrew/perls/perl-5.22.1/man
    PERLBREW_PATH=/Users/dur-randir/perlbrew/bin:/Users/dur-randir/perlbrew/perls/perl-5.22.1/bin
    PERLBREW_PERL=perl-5.22.1
    PERLBREW_ROOT=/Users/dur-randir/perlbrew
    PERLBREW_VERSION=0.78
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/zsh

```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Fri\, 01 Dec 2017 15&#8203;:43&#8203;:51 GMT\, randir wrote&#8203;:
> This is a bug report for perl from sergey\.aleynikov@&#8203;gmail\.com\,
> generated with the help of perlbug 1\.41 running under perl 5\.27\.7\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> Recent commit 184f90dc410cf3e66a5e682337ae54b3c9f9311f
> Author&#8203;: Tony Cook \<tony@&#8203;develop\-help\.com>
> Date&#8203;:   Fri Dec 1 01&#8203;:05&#8203;:07 2017 \+1100
> 
> \(perl \#132506\) deal with un\-/partly implemented \*at\(\) functions
> 
> broke the following test&#8203;:
> 
> % \./perl \./harness run/switches\.t
> run/switches\.t \.\. 47/137 \# Failed test 130 \- fail final rename at
> \./test\.pl line 1074
> \#      got 'Cannot complete in\-place edit of tmpinplace/foo&#8203;: failed to
> rename work file \\'tmpinplace/SrDX3NRv\\' to \\'tmpinplace/foo\\'&#8203;:
> Permission denied at \- line 5\, \<> line 1\.'
> \# expected /\(?^&#8203;:Can't rename\)/
> \# PROG&#8203;:
> \# @&#8203;ARGV = \("tmpinplace/foo"\);
> \# $^I = "";
> \# while \(\<>\) \{
> \#   chmod 0500\, "tmpinplace";
> \#   print;
> \# \}
> \# print "ok\\n";
> \# STATUS&#8203;: 3328
> run/switches\.t \.\. Failed 1/137 subtests
>         \(less 6 skipped subtests&#8203;: 130 okay\)
> 
> Test Summary Report
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> run/switches\.t \(Wstat&#8203;: 0 Tests&#8203;: 137 Failed&#8203;: 1\)
>   Failed test&#8203;:  130
> Files=1\, Tests=137\,  1 wallclock secs \( 0\.05 usr  0\.01 sys \+  0\.53
> cusr  0\.60 csys =  1\.19 CPU\)
> Result&#8203;: FAIL
> 

Apparently that's not the only case where t/run/switches\.t is broken&#8203;:

\#\#\#\#\#
git show a06de4dcf2448ead9a5f5dbd87366727ebd91aca
commit a06de4dcf2448ead9a5f5dbd87366727ebd91aca
Author&#8203;: Steve Hay \<steve\.m\.hay@&#8203;googlemail\.com>
Date&#8203;:   Thu Nov 30 09&#8203;:05&#8203;:45 2017 \+0000

&nbsp;    Fix Windows build following commit 184f90dc41
&nbsp;    
&nbsp;    But t/run/switches\.t test 131 still fails\, with perl\.exe crashing\.

<details><summary>Inline Patch</summary>

```diff
diff --git a/doio.c b/doio.c
index 565f896..9f76c10 100644
--- a/doio.c
+++ b/doio.c
@@ -1267,7 +1267,7 @@ S_dir_unchanged(pTHX_ const char *orig_pv, MAGIC *mg) {
     if (!PERL_FILE_IS_ABSOLUTE(orig_pv)
         && PerlLIO_stat(SvPVX(*temp_psv), &statbuf) < 0) {
         Perl_croak(aTHX_ "Cannot complete in-place edit of %" SVf ": %s",
-                   *orig_psv,
+                   orig_pv,
                    "Work file is missing - did you change directory?");
     }
 #endif

```
</details>
#####



\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout

On Fri\, 01 Dec 2017 07&#8203;:43&#8203;:51 \-0800\, randir wrote&#8203;:
> This is a bug report for perl from sergey\.aleynikov@&#8203;gmail\.com\,
> generated with the help of perlbug 1\.41 running under perl 5\.27\.7\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> Recent commit 184f90dc410cf3e66a5e682337ae54b3c9f9311f
> Author&#8203;: Tony Cook \<tony@&#8203;develop\-help\.com>
> Date&#8203;:   Fri Dec 1 01&#8203;:05&#8203;:07 2017 \+1100
> 
> \(perl \#132506\) deal with un\-/partly implemented \*at\(\) functions
> 
> broke the following test&#8203;:
> 
> % \./perl \./harness run/switches\.t
> run/switches\.t \.\. 47/137 \# Failed test 130 \- fail final rename at
> \./test\.pl line 1074
> \#      got 'Cannot complete in\-place edit of tmpinplace/foo&#8203;: failed to
> rename work file \\'tmpinplace/SrDX3NRv\\' to \\'tmpinplace/foo\\'&#8203;:
> Permission denied at \- line 5\, \<> line 1\.'
> \# expected /\(?^&#8203;:Can't rename\)/
> \# PROG&#8203;:
> \# @&#8203;ARGV = \("tmpinplace/foo"\);
> \# $^I = "";
> \# while \(\<>\) \{
> \#   chmod 0500\, "tmpinplace";
> \#   print;
> \# \}
> \# print "ok\\n";
> \# STATUS&#8203;: 3328
> run/switches\.t \.\. Failed 1/137 subtests
>         \(less 6 skipped subtests&#8203;: 130 okay\)

Yves Orton fixed the test with this commit&#8203;:

https://perl5.git.perl.org/perl.git/commitdiff/8779d80ae

But I don’t know whether the failure was platform\-specific\, and whether it will now start failing on platforms where it previously passed\.  Will Tony Cook please speak up?

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
### From @dur-randir

On Fri\, 01 Dec 2017 12&#8203;:34&#8203;:10 \-0800\, sprout wrote&#8203;:
> Yves Orton fixed the test with this commit&#8203;:
> 
> https://perl5.git.perl.org/perl.git/commitdiff/8779d80ae

I can confirm that\, at least on Darwin\, this is fixed as of now\.

**Comment by p5pRT** [comment]
### From @tonycoz

On Fri\, 01 Dec 2017 12&#8203;:34&#8203;:10 \-0800\, sprout wrote&#8203;:
> Yves Orton fixed the test with this commit&#8203;:
> 
> https://perl5.git.perl.org/perl.git/commitdiff/8779d80ae
> 
> But I don’t know whether the failure was platform\-specific\, and
> whether it will now start failing on platforms where it previously
> passed\.  Will Tony Cook please speak up?

The test was skipped on Win32\, which is where I was working when I broke this\.

Yves patch appears to have fixed the only problem\.  The tests pass on NetBSD 6 \(which led to the original breakage\) and Win32 \(which led to the new breakage\)\.

Closing\.

Tony

**Comment by p5pRT** [comment]
@tonycoz - Status changed from 'open' to 'resolved'
